src/bin/jcli_app/transaction/mk_witness.rs

```diff
@@ -62,50 +45,44 @@ pub enum WitnessType {
     OldUTxO,
     Account,
 }
-impl std::fmt::Display for WitnessType {
-    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
-        match self {
-            WitnessType::UTxO => write!(f, "utxo"),
-            WitnessType::OldUTxO => write!(f, "legacy-utxo"),
-            WitnessType::Account => write!(f, "account"),
-        }
-    }
-}
+
 impl std::str::FromStr for WitnessType {
-    type Err = ParseWitnessTypeError;
+    type Err = &'static str;
+
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         match s {
             "utxo" => Ok(WitnessType::UTxO),
             "legacy-utxo" => Ok(WitnessType::OldUTxO),
             "account" => Ok(WitnessType::Account),
-            _ => Err(ParseWitnessTypeError::Invalid),
+            _ => Err("Invalid witness type, expected `utxo', `legacy-utxo' or `account'"),
         }
     }
 }
 
 impl MkWitness {
-    fn secret<A: AsymmetricKey>(&self) -> Result<SecretKey<A>, MkWitnessError> {
-        let mut bech32_str = String::new();
-        io::open_file_read(&self.secret)
-            .unwrap()
-            .read_to_string(&mut bech32_str)?;
-        Ok(SecretKey::try_from_bech32_str(&bech32_str)?)
+    fn secret<A: AsymmetricKey>(&self) -> Result<SecretKey<A>, Error> {
+        let bech32_str =
+            common::read_line(&self.secret).map_err(|source| Error::SecretFileReadFailed {
+                source,
+                path: common::path_to_path_buf(&self.secret),
+            })?;
+        SecretKey::try_from_bech32_str(&bech32_str).map_err(|source| Error::SecretFileMalformed {
+            source,
+            path: common::path_to_path_buf(&self.secret),
+        })
     }
 
-    pub fn exec(self) -> Result<(), MkWitnessError> {
+    pub fn exec(self) -> Result<(), Error> {
         let witness = match self.witness_type {
             WitnessType::UTxO => {
                 let secret_key = self.secret()?;
                 Witness::new_utxo(&self.genesis_block_hash, &self.transaction_id, &secret_key)
             }
-            WitnessType::OldUTxO => {
-                // let secret_key = self.secret()?;
-                unimplemented!()
```

Comment:
How about this: we keep the clean error and add below:
```rust
// TODO unimplemented!()
```
This way it will be obvious and easily searchable for developers and will still provide clean error for user.